### PR TITLE
Enable importing TinyYOLOv2

### PR DESCRIPTION
**Pull request by mbencer**

### CODE REVIEW

File: ngraph/src/ngraph/validation_util.hpp [hunk 1]
```diff
@@ -89,6 +89,15 @@ namespace ngraph
                                  const PartialShape& gamma_shape,
                                  const PartialShape& beta_shape);
 
+    NGRAPH_API
+    bool try_apply_infer_auto_padding(const PartialShape& image_shape,
```
Comment (postrational) [inline ngraph/src/ngraph/validation_util.hpp:93]: There is something wrong with the name of this function. I can't tell what it's supposed to do.
Please change name and add some documentation.

File: ngraph/test/type_prop/convolution.cpp [hunk 2]
```diff
@@ -2884,6 +2884,64 @@ TEST(type_prop, conv_v1_partial_rank)
...
+    Strides dilations{1, 1};
+    const auto auto_pad = op::PadType::SAME_LOWER;
+
+    auto data_batch = make_shared<op::Parameter>(element::f32, data_batch_shape);
+    auto filters = make_shared<op::Parameter>(element::f32, filters_shape);
+
+    auto conv = make_shared<op::v1::Convolution>(
+        data_batch, filters, strides, pads_begin, pads_end, dilations, auto_pad);
+
+    ASSERT_TRUE(conv->get_output_partial_shape(0).same_scheme(PartialShape{1, 1, 5, 5}));
```
Comment (arogowie-intel) [inline ngraph/test/type_prop/convolution.cpp:2903]: You can also check here the padding values.

File: ngraph/test/type_prop/convolution.cpp [hunk 3]
```diff
@@ -2884,6 +2884,64 @@ TEST(type_prop, conv_v1_partial_rank)
...
+    Strides dilations{1, 1};
+    const auto auto_pad = op::PadType::SAME_LOWER;
+
+    auto data_batch = make_shared<op::Parameter>(element::f32, data_batch_shape);
+    auto filters = make_shared<op::Parameter>(element::f32, filters_shape);
+
+    auto conv = make_shared<op::v1::Convolution>(
+        data_batch, filters, strides, pads_begin, pads_end, dilations, auto_pad);
+
+    ASSERT_TRUE(conv->get_output_partial_shape(0).same_scheme({Dimension::dynamic(), 1, 5, 5}));
```
Comment (arogowie-intel) [inline ngraph/test/type_prop/convolution.cpp:2922]: As above.

File: ngraph/test/type_prop/convolution.cpp [hunk 4]
```diff
@@ -2884,6 +2884,64 @@ TEST(type_prop, conv_v1_partial_rank)
...
+
+TEST(type_prop, conv_v1_partial_auto_padding_same_nc_dims_dynamic)
+{
+    const PartialShape data_batch_shape{Dimension::dynamic(), Dimension::dynamic(), 5, 5};
+    const PartialShape filters_shape{1, 1, 3, 3};
+    Strides strides{1, 1};
+    CoordinateDiff pads_begin{0, 0};
+    CoordinateDiff pads_end{0, 0};
+    Strides dilations{1, 1};
+    const auto auto_pad = op::PadType::SAME_LOWER;
```
Comment (arogowie-intel) [inline ngraph/test/type_prop/convolution.cpp:2914]: Please add one test with `SAME_UPPER`.

File: ngraph/test/type_prop/max_pool.cpp [hunk 5]
```diff
@@ -588,3 +588,56 @@ TEST(type_prop, max_pool_partial_rank_static_dynamic_padded_window_not_too_big)
...
+
+TEST(type_prop, max_pool_auto_padding_nc_dims_dynamic)
+{
+    const PartialShape arg_shape{Dimension::dynamic(), Dimension::dynamic(), 32, 32};
+    const Strides strides{1, 1};
+    const Shape pads_begin{0, 0};
+    const Shape pads_end{0, 0};
+    const Shape kernel_shape{2, 2};
+    const auto rounding_mode = op::RoundingType::FLOOR;
+    const auto auto_pad = op::PadType::SAME_LOWER;
```
Comment (arogowie-intel) [inline ngraph/test/type_prop/max_pool.cpp:617]: Similarly as for convolution, would be good to have test also for `SAME_UPPER` and to check also inferred padding values.

**Comment by jane-intel** [comment]
Please apply the change carefully to all the operations that use auto_pad: deformable convolution, binary convolution and all poolings.

**Comment by mbencer** [comment]
> Please apply the change carefully to all the operations that use auto_pad: deformable convolution, binary convolution and all poolings.

You are right, there was created task for it as separate story: CVS-34151

File: ngraph/test/type_prop/convolution.cpp (hunk 2 above)
Comment (mbencer) [reply ngraph/test/type_prop/convolution.cpp:2576]: done

File: ngraph/test/type_prop/convolution.cpp (hunk 4 above)
Comment (mbencer) [reply ngraph/test/type_prop/convolution.cpp:2589]: done - `conv_v1_partial_auto_padding_same_nc_dims_dynamic_same_upper`

File: ngraph/test/type_prop/convolution.cpp (hunk 3 above)
Comment (mbencer) [reply ngraph/test/type_prop/convolution.cpp:2597]: done

File: ngraph/test/type_prop/max_pool.cpp (hunk 5 above)
Comment (mbencer) [reply ngraph/test/type_prop/max_pool.cpp:619]: done

**Pull request merged**